### ISSUE TITLE
Failed building cabal-macosx with cabal new-build

### ISSUE THREAD

**Issue by rizary** [danfran/cabal-macosx#10]
When I build one of my project using cabal new-build, I got an error on building cabal-macosx. So in that case, I do cabal install cabal-macosx with cabal head and got the following error.

```
[4 of 7] Compiling Distribution.MacOSX.Internal ( Distribution/MacOSX/Internal.hs, dist/build/macosx-app/macosx-app-tmp/Distribution/MacOSX/Internal.o )

Distribution/MacOSX/Internal.hs:43:31: error:
    • Couldn't match type ‘Distribution.Types.UnqualComponentName.UnqualComponentName’
                     with ‘[Char]’
      Expected type: String
        Actual type: Distribution.Types.UnqualComponentName.UnqualComponentName
    • In the first argument of ‘MacApp’, namely ‘(exeName x)’
      In the expression:
        MacApp (exeName x) Nothing Nothing [] [] DoNotChase
      In an equation for ‘mkDefault’:
          mkDefault x = MacApp (exeName x) Nothing Nothing [] [] DoNotChase

Distribution/MacOSX/Internal.hs:47:52: error:
    • Couldn't match type ‘[Char]’
                     with ‘Distribution.Types.UnqualComponentName.UnqualComponentName’
      Expected type: Distribution.Types.UnqualComponentName.UnqualComponentName
        Actual type: String
    • In the second argument of ‘(==)’, namely ‘appName app’
      In the expression: exeName e == appName app
      In the first argument of ‘any’, namely
        ‘(\ e -> exeName e == appName app)’
cabal: Leaving directory '/var/folders/s_/95c8yrm94h1chy1jmg31y18w0000gn/T/cabal-tmp-65581/cabal-macosx-0.2.4.0'
World file is already up to date.
CallStack (from HasCallStack):
  die', called at ./Distribution/Client/Install.hs:994:15 in cabal-install-2.1.0.0-inplace:Distribution.Client.Install
cabalH: Error: some packages failed to install:
cabal-macosx-0.2.4.0-8qLIpmkXRz7Kmwdv2gG1Nr failed during the building phase.
The exception was:
ExitFailure 1
```

**Comment by tkemps** [comment]
I get the same error for cabal-macosx-0.2.4.0 with GHC 8.0.2 when I use the recipe at https://wiki.haskell.org/WxHaskell/Mac. I am on macOS Sierra 10.12.6.

**Comment by danfran** [comment]
I am about to add the GHC 8.0.2 to travis and merging again the fixes for the proposed fixes.

**Comment by danfran** [comment]
I have release a new 0.2.4.1. I have fixed the build (test included) for Cabal 2.0. Also, I have removed the versions for 7.0.4 and 7.2.2 from Travis. Let me know if there are problems or not. In case I will close this ticket.
